### DISCUSSION TITLE
Another project you could reference for flash is gnash from GNU

### DISCUSSION THREAD

**Discussion by l12436** [ruffle-rs/ruffle#1359] [Ideas]
https://www.gnu.org/software/gnash/

you may get some useful implement idea from gnash maybe

**Comment by soulthunk** [comment]
Looks cool, it could help with some older flashes.

**Comment by ROBERT-MCDOWELL** [comment]
gnash is abandoned since long time! not fully API compatible.

**Reply by l12436** [reply]
yes, but ruffle also did not fully support in old flash, maybe some could referenced in gnash XD

**Comment by ROBERT-MCDOWELL** [comment]
If you mean by "old" AVM1 so yes Ruffle support it, but not work in progress! you are welcome to contribute so it will speed up the full API coverage

**Reply by l12436** [reply]
not criticized about ruffle. I am sorry if you feel offended :)
I am looking forward about this project, the original flash player also has some bug in it.
I know nothing about flash and rust XD

**Comment by ROBERT-MCDOWELL** [comment]
lol, not offensed at all! I just say what it is...
btw, maybe it's time to learn rust, it's like javascript but more clean. It's worth to learn it as it will help this project and future ones to evolve and develop faster.
